### PR TITLE
fix(DSC): fix dsc product id

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
fix dsc product id, get the product id from CBC, not a fixed value
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix dsc product id, get the product id from CBC, not a fixed value
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dsc/ TESTARGS='-run TestAccDscInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dsc/ -v -run TestAccDscInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDscInstance_basic
=== PAUSE TestAccDscInstance_basic
=== CONT  TestAccDscInstance_basic
--- PASS: TestAccDscInstance_basic (88.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dsc       88.216s
```

### CODE REVIEW

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go [hunk 1]
```diff
@@ -293,18 +290,119 @@ func buildCreateDscInstanceRequestBodyProductInfos(d *schema.ResourceData) []map
...
+	region := cfg.GetRegion(d)
+	productInfos := make([]map[string]interface{}, len(rst))
+	period := d.Get("period").(int)
+	periodUnit := d.Get("period_unit").(string)
+	for i, product := range rst {
+		productInfos[i] = buildProductInfos(cfg.GetRegion(d), periodUnit, strconv.Itoa(i+1), period, product)
+	}
+	products, err := getOrderProductId(cfg, region, productInfos)
+	if err != nil {
+		return fmt.Errorf("error getting DSC order product id infos: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]:
```suggestion
		return fmt.Errorf("error getting DSC order product ID infos: %s", err)
```
Please check the other parts.

File: huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go [hunk 2]
```diff
@@ -293,18 +290,119 @@ func buildCreateDscInstanceRequestBodyProductInfos(d *schema.ResourceData) []map
...
+	return rst, nil
+}
 
+func addProductIdToProductInfo(d *schema.ResourceData, cfg *config.Config, rst []map[string]interface{}) error {
+	region := cfg.GetRegion(d)
+	productInfos := make([]map[string]interface{}, len(rst))
+	period := d.Get("period").(int)
+	periodUnit := d.Get("period_unit").(string)
+	for i, product := range rst {
+		productInfos[i] = buildProductInfos(cfg.GetRegion(d), periodUnit, strconv.Itoa(i+1), period, product)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]: it's better to add some comment about the relation between id and index
```suggestion
		// add some comment
		productID := strconv.Itoa(i+1)
		productInfos[i] = buildProductInfos(region, periodUnit, productID, period, product)
```

Comment (houpeng80) [reply huaweicloud/services/dsc/resource_huaweicloud_dsc_instance.go]: done

**Comment by houpeng80** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dsc/ TESTARGS='-run TestAccDscInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dsc/ -v -run TestAccDscInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDscInstance_basic
=== PAUSE TestAccDscInstance_basic
=== CONT  TestAccDscInstance_basic
--- PASS: TestAccDscInstance_basic (93.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dsc       93.752s

**Pull request merged**